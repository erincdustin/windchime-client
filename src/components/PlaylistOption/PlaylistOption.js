import React from 'react';

function PlaylistOption(props) {
  let option='';
  if (props.genreOption){
    option = props.genreOption
  } if (props.topArtists){
    option = 'Top Artists'
  }

  return (
    <div>
      <h3 className="ribbon">Playlist Options<span className="ribbon-text">{option}</span></h3>
    </div>
  );
}

export default PlaylistOption;