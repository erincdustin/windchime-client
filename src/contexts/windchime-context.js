import React, { Component } from 'react'

const WindChimeContext = React.createContext({
    error: '',
    songs: null,
    weather: null,
    genreChoice: null,
    id: null,
    artists: null,
    topArtists: null,
    playlistId: null,
    userPlaylists: null,
    snapshot: null,
    targetEnergy: null,
    targetValence: null,
    targetTempo: null,
    targetPopularity: null,
    setError: () => { },
    clearError: () => { },
    setSongs: () => { },
    setWeather: () => { },
    setGenreChoice: () => { },
    setId: () => { },
    setArtists: () => { },
    setTopArtists: () => { },
    setPlaylistId: () => { },
    setUserPlaylists: () => { },
    setSnapshot: () => { },
    setTargetEnergy: () => { },
    setTargetValence: () => { },
    setTargetTempo: () => { },
    setTargetPopularity: () => { },
})

export default WindChimeContext;

export class WindChimeProvider extends Component {
    constructor(props) {
        super(props)
        const state = {
            error: '',
            songs: null,
            weather: null,
            genreChoice: null,
            id: null,
            artists: null,
            topArtists: null,
            userPlaylists: null,
            playlistId: null,
            snapshot: null,
            targetEnergy: null,
            targetValence: null,
            targetTempo: null,
            targetPopularity: null,
            setError: this.setError,
            clearError: this.clearError,
            setSongs: this.setSongs,
            setWeather: this.setWeather,
            setGenreChoice: this.setGenreChoice,
            setId: this.setId,
            setArtists: this.setArtists,
            setTopArtists: this.setTopArtists,
            setPlaylistId: this.setPlaylistId,
            setUserPlaylists: this.setUserPlaylists,
            setSnapshot: this.setSnapshot,
            setTargetEnergy: this.setTargetEnergy,
            setTargetValence: this.setTargetValence,
            setTargetTempo: this.setTargetTempo,
            setTargetPopularity: this.setTargetPopularity
          }   
        this.state = state; 
    }

    setError = error => {
      console.error(error)
      this.setState({ error })
    }

    clearError = () => {
      this.setState({ error: '' })
    }

    setUserPlaylists = (userPlaylists) => {
      this.setState({ userPlaylists })
    }

    setSongs = (songs) => {
      this.setState({ songs })
    }

    setWeather = (weather) => {
      this.setState({ weather })
    }

    setGenreChoice = (genreChoice) => {
      this.setState({ genreChoice })
    }

    setId = (id) => {
      this.setState({ id })
    }

    setArtists = (artists) => {
      this.setState({ artists })
    }

    setTopArtists = (topArtists) => {
      this.setState({ topArtists })
    }

    setPlaylistId = (playlistId) => {
      this.setState({ playlistId })
    }

    setSnapshot = (snapshot) => {
      this.setState({ snapshot })
    }

    setTargetEnergy = (targetEnergy) => {
      this.setState({ targetEnergy })
    }

    setTargetValence = (targetValence) => {
      this.setState({ targetValence })
    }

    setTargetTempo = (targetTempo) => {
      this.setState({ targetTempo })
    }

    setTargetPopularity = (targetPopularity) => {
      this.setState({ targetPopularity })
    }

    render() {
        return (
            <WindChimeContext.Provider value={this.state}>
                {this.props.children}
            </WindChimeContext.Provider>
        )
    }
}
