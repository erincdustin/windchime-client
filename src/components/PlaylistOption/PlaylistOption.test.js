import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import { WindChimeProvider } from '../../contexts/windchime-context';
import { library } from '@fortawesome/fontawesome-svg-core'
import { faWind } from '@fortawesome/free-solid-svg-icons'
import PlaylistOption from './PlaylistOption';

library.add(faWind)

describe(`PlaylistOption`, () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(<WindChimeProvider>
          <BrowserRouter>
            <PlaylistOption />
          </BrowserRouter>
        </WindChimeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
