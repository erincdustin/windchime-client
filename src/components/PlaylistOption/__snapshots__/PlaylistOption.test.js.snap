// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistOption renders as expected 1`] = `
<div>
  <div
    className="ribbon two"
  >
    <div
      className="ribbon-header"
    >
      Playlist Options
    </div>
    <span
      className="ribbon-text"
    >
      <span
        className="hidden"
      >
        my top artists
      </span>
      <button
        className="btn weather"
        onClick={[Function]}
      >
        Change
      </button>
    </span>
  </div>
</div>
`;
